lib/DxilContainer/DxcContainerBuilder.cpp

```diff
@@ -98,12 +60,13 @@ HRESULT STDMETHODCALLTYPE DxcContainerBuilder::AddPart(_In_ UINT32 fourCC, _In_
     IFTBOOL(pSource != nullptr && !IsDxilContainerLike(pSource->GetBufferPointer(),
       pSource->GetBufferSize()),
       E_INVALIDARG);
-    // Only allow adding private data, debug info name and root signature for now
-    IFTBOOL(
-        fourCC == DxilFourCC::DFCC_RootSignature || 
+    // You can only add debug info, debug info name, rootsignature, or private data blob
```

Comment:
Should probably update the comment with reflection (DFCC_ShaderStatistics).  And I think we should allow adding the `DFCC_ShaderHash` as well.  We would need to update the RemovePart with the corresponding change as well.  And we should list the valid parts to use with this API in the header.